### PR TITLE
Add esp32 and esp32c3 VSCode debugging section

### LINKED ISSUE #29: How to debug with vscode
Can anyone detail the steps required to setup debugging?

I'm working with the ESP32-C3 and can flash with command line using "cargo espflash --monitor COM8" although would be good if this could be done through vscode, and optionally launch the debugger.

### PR DESCRIPTION
**Pull request by JurajSadel**
closes #29

### CODE REVIEW

File: src/tooling/debugging/vscode-debugging.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    }
+```
+
+# ESP32-C3
+
+Older versions with **revision < 3** **doesn't** have built-in JTAG interface.
+
+ESP32-C3 with **revision 3** **does** have a built-in JTAG interface and you don't have to connect an external device to be able to debug. To get the chip revision, run the `cargo espflash board-info` command.
+
+## HW setup
```
Comment (jessebraham) [inline src/tooling/debugging/vscode-debugging.md:76]:
```suggestion
## Hardware Setup
```

File: src/tooling/debugging/vscode-debugging.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+|   ESP32-C3 Pin   |   JTAG Signal   |
+| :--------------: | :-------------: |
+| MTDO/GPIO7       |      TDO        |
+| MTDI/GPIO5       |      TDI        |
+| MTCK/GPIO6       |      TCK        |
+| MTMS/GPIO4       |      TMS        |
+| 3V3              |      VJTAG      |
+| GND              |      GND        |
+
+Now, ESP32-C3 does have a JTAG interface, and both devices, ESP32-C3 and ESP-Prog can be plugged into your machine. The HW part is ready.
```
Comment (jessebraham) [inline src/tooling/debugging/vscode-debugging.md:91]: Same as in the previous comment.

**Review by jessebraham — CHANGES_REQUESTED** [review]
Looks good overall, thanks for writing all these out! I've requested a handful of small changes be made if you don't mind, after that I'm happy to merge.

**Comment by JurajSadel** [comment]
Thanks @jessebraham !

File: src/tooling/debugging/vscode-debugging.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+| GND           |      GND        |
+
+
+**Note**: On Windows `USB Serial Converter A 0403 6010 00` driver should be WinUSB.
+
+## Set up VSCode
+
+1. Install [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) extension for VScode.
+2. Create the `.vscode/launch.json` file in the project tree you want to debug. [This](https://github.com/esp-rs/esp32-hal/blob/master/.vscode/launch.json) can be used as a template file.
+3. Update **executable**, **svdFile** paths and **toolchainPrefix** field.
```
Comment (MabezDev) [inline src/tooling/debugging/vscode-debugging.md:28]: They also need to change serverpath to the xtensa enabled espressif one

File: src/tooling/debugging/vscode-debugging.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+| GND              |      GND        |
+
+
+**Note**: On Windows `USB Serial Converter A 0403 6010 00` driver should be WinUSB.
+
+## Set up VSCode
+
+1. Install [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) extension for VScode.
+2. Create the `.vscode/launch.json` file in the project tree you want to debug. [This](https://github.com/esp-rs/esp32-hal/blob/master/.vscode/launch.json) can be used as a template file.
+3. Update **executable**, **svdFile** paths and **toolchainPrefix** field.
```
Comment (MabezDev) [inline src/tooling/debugging/vscode-debugging.md:97]: Ditto, serverpath needs changing too

File: src/tooling/debugging/vscode-debugging.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+```json
```
Comment (MabezDev) [inline src/tooling/debugging/vscode-debugging.md:98]: Consider using `jsonc` here as the code formatter, should get rid of those nasty red lines on the comments :)

**Review by MabezDev — CHANGES_REQUESTED** [review]

File: src/tooling/debugging/vscode-debugging.md (hunk 3 above)
Comment (JurajSadel) [reply src/tooling/debugging/vscode-debugging.md]: That's good catch, thanks @MabezDev !

**Review by MabezDev — APPROVED** [review]
LGTM! We might have to change the title of the section if we ever have a probe-rs vscode configuration, but we'll cross that bridge when we get to it :D. Great work @JurajSadel!

**Review by jessebraham — APPROVED** [review]
Thanks for making those changes @JurajSadel, and thanks for the additional review @MabezDev!

**Pull request merged**